samples/scripts/externalclient-setup.js

```diff
@@ -0,0 +1,299 @@
+// the test script.
+//
+// It is safe to run the setup script again. A new pool is created and new
+// audiences are printed. If run multiple times, it is advisable to delete
+// unused pools. Note that deleted pools are soft deleted and may remain for
+// a while before they are completely deleted. The old pool ID cannot be used
+// in the meantime.
+
+const fs = require('fs');
+const {promisify} = require('util');
+const {GoogleAuth} = require('google-auth-library');
+
+const readFile = promisify(fs.readFile);
+
+/**
+ * Generates a random string of the specified length, optionally using the
+ * specified alphabet.
+ *
+ * @param {number} length The length of the string to generate.
+ * @return {string} A random string of the provided length.
+ */
+const generateRandomString = length => {
+  const chars = [];
+  const allowedChars = 'abcdefghijklmnopqrstuvwxyz0123456789';
+  while (length > 0) {
+    chars.push(
+      allowedChars.charAt(Math.floor(Math.random() * allowedChars.length))
+    );
+    length--;
+  }
+  return chars.join('');
+};
+
+/**
+ * Creates a workload identity pool with an OIDC provider which will accept
+ * Google OIDC tokens generated from the current service account where the token
+ * will have sub as the service account client ID and the audience as the
+ * created identity pool STS audience.
+ * The steps followed here mirror the instructions for configuring federation
+ * with an OIDC provider illustrated at:
+ * https://cloud.google.com/iam/docs/access-resources-oidc
+ * This will also create an AWS provider in the same workload identity pool
+ * using the AWS account ID and AWS ARN role name provided.
+ * The steps followed here mirror the instructions for configuring federation
+ * with an AWS provider illustrated at:
+ * https://cloud.google.com/iam/docs/access-resources-aws
+ * @param {Object} config An object containing additional data needed to
+ *   configure the external account client setup.
+ * @return {Promise<Object>} A promise that resolves with an object containing
+ *   the STS audience corresponding with the generated workload identity pool
+ *   OIDC provider and AWS provider.
+ */
+async function main(config) {
+  let response;
+  const keyFile = process.env.GOOGLE_APPLICATION_CREDENTIALS;
+  if (!process.env.GOOGLE_APPLICATION_CREDENTIALS) {
+    throw new Error('No GOOGLE_APPLICATION_CREDENTIALS env var is available');
+  }
+  const keys = JSON.parse(await readFile(keyFile, 'utf8'));
+  const suffix = generateRandomString(10);
+  const poolId = `pool-${suffix}`;
+  const oidcProviderId = `oidc-${suffix}`;
+  const awsProviderId = `aws-${suffix}`;
+  const projectId = keys.project_id;
+  const clientEmail = keys.client_email;
+  const sub = keys.client_id;
+  const auth = new GoogleAuth({
+    scopes: 'https://www.googleapis.com/auth/cloud-platform',
+  });
+
+  // TODO: switch to using IAM client SDK once v1 API has all the v1beta
```

Comment:
Please make sure we're filing issues on GitHub for any TODOs so we don't lose them.

Reply:
Done: https://github.com/googleapis/google-auth-library-nodejs/issues/1132